sql: assorted refactorings for upcoming schema changer changes

andreimatei opened the pull request:
See the individual commits.

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/6160)

<!-- Reviewable:end -->

andreimatei committed:
sql: add a version of GetTableDesc that doesn't error if the table is not found

andreimatei committed:
sql: Make Publish() return the updated version of the descriptor.

This will be used in the next commit by the schema changer to figure
out if the descriptor whose version it's upping is to be deleted.

vivekmenezes commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Review status: 0 of 21 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[sql/drop.go, line 41 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFk7R9_vCutSquFFrKt:-KFk7R9_vCutSquFFrKu:1861458350) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/drop.go#L41)):*
what about pErr here?

---

*[sql/lease.go, line 293 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFk8jLYF2tgRHvVWB3C:-KFk8jLYF2tgRHvVWB3D:800113905) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L293)):*
why do we need this?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFkACWp6B_sQsUSG_Xf:1417088721)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
Review status: 0 of 21 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[sql/drop.go, line 41 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFk7R9_vCutSquFFrKt:-KFkcusk-fcZEvGagFhZ:-301364910) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/drop.go#L41)):*
you're right, done

---

*[sql/lease.go, line 293 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFk8jLYF2tgRHvVWB3C:-KFkdRL5-t9BaSXOo8dz:-2054085178) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L293)):*
I think it's more readable this way, get the no error case out early. Otherwise the logic below gets hard to trace I think. Plus it's not intuitive (at least not to me) that GetDetail works if pErr is nil.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 21 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[sql/lease.go, line 293 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFk8jLYF2tgRHvVWB3C:-KFnutcXz1OoFVEHUx9V:1598814817) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L293)):*
I think this code is much more readable as a select. Wanna make that change?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

tamird commented:
Second commit just has a bunch of unrelated changes that aren't called out in the commit message at all - was that intended?

---

Reviewed 2 of 2 files at r1, 17 of 17 files at r2, 4 of 4 files at r3, 5 of 5 files at r4, 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 25 unresolved discussions, all commit checks successful.

---

*[server/admin_test.go, line 328 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo938LuOfb0nB5g1Du:-KFo938LuOfb0nB5g1Dv:-344819823) ([raw file](https://github.com/cockroachdb/cockroach/blob/86354684a992a0970e7cf46962daea0ca6307eda/server/admin_test.go#L328)):*
how come "table" is no longer being searched for?

---

*[server/admin_test.go, line 344 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo9C0cAs7KWSpku4oQ:-KFo9C0cAs7KWSpku4oR:1851594930) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/server/admin_test.go#L344)):*
picking nits now but why `%+v`?

---

*[sql/create.go, line 104 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo7X3wBTzejvFBPmQu:-KFo7X3wBTzejvFBPmQv:-1496732755) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/create.go#L104)):*
another refactor which is not mentioned in the commit message

---

*[sql/database.go, line 30 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo65UUSC1FDiVYjfJZ:-KFo65UV9X2akfoxPteM:1446537092) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L30)):*
why is this not an `error`?

---

*[sql/database.go, line 94 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6JvzWwuhHb7jtXFC:-KFo6JvzWwuhHb7jtXFD:-885450162) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L94)):*
this kind of method proliferation is no bueno in my opinion. I would suggest changing this API to take a `descriptor` argument (as `getDescriptor` does) and return `(bool, roachpb.Error)`.

---

*[sql/database.go, line 142 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo7RvIgxQ42CJXIDWm:-KFo7RvIgxQ42CJXIDWn:2046355521) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L142)):*
this refactor seems to have nothing to do with what's described in the commit message.

---

*[sql/database.go, line 143 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6jrqNZgp87e-7MMB:-KFo6jrqNZgp87e-7MMC:-505040402) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L143)):*
please follow the style guide.

---

*[sql/database.go, line 148 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6mAyY86h9ggdT9fW:-KFo6mAyY86h9ggdT9fX:-768350736) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L148)):*
please avoid naked returns.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFo6wl3lW3PxOVd9TfT:453912664) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
why is this true? in general, `true, <non-nil>` seems like a nonsense return tuple.

---

*[sql/lease.go, line 227 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo9kxqPglbfr9PyXfO:-KFo9kxqPglbfr9PyXfP:-505040402) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L227)):*
please follow the style guide.

---

*[sql/lease.go, line 243 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoAIDA3W_5j3tv4B13:-KFoAIDA3W_5j3tv4B14:2141249836) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L243)):*
any reason not to hoist this allocation all the way out of the retry loop?

---

*[sql/structured.go, line 640 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8DscxhwvdL-aMy5h:-KFo8DscxhwvdL-aMy5i:1942985053) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L640)):*
again, maybe give this a more meaningful name? It seems like this is really "getNextMutationID" and the increment is just a side effect.

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFo85LCUOUw57aX_phG:2107166194) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
"setUpVersion" is really not better than `desc.UpVersion = true` at all. Perhaps you could give this method a name that is more semantically meaningful?

---

*[sql/table.go, line 50 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8L1lTbqX7tGcwY0h:-KFo8L1lTbqX7tGcwY0i:1666505909) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L50)):*
this return type should be `error`

---

*[sql/table.go, line 227 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8PdXRsSDfIAiGib9:-KFo8PdXRsSDfIAiGibA:-1491170426) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L227)):*
style guide.

---

*[sql/table.go, line 232 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8REFWeeBZSJYDQZH:-KFo8REFWeeBZSJYDQZI:-1478024119) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L232)):*
avoid naked return

---

*[sql/table.go, line 260 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8Vs4W-mdWDdjBLMN:-KFo8Vs4W-mdWDdjBLMO:941789391) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L260)):*
ditto re method proliferation.

---

*[sql/truncate.go, line 57 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8dtvnRSbmWew19aX:-KFo8dtvnRSbmWew19aY:-930429378) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/truncate.go#L57)):*
another random refactor not mentioned anywhere. also this one doesn't seem to actually save anything =/

---

*[sql/parser/expr.go, line 482 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo5hKJlLnoWK895xWU:-KFo5hKJlLnoWK895xWV:1277541042) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L482)):*
does this need to be exported? it's only used for tests in `package parser`, though I'd honestly prefer that this method be replaced with a deepCopy method with a value receiver as suggested below.

---

*[sql/parser/expr.go, line 491 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo5OUyckwv9FWYYrE1:-KFo5OUzXmwFdF2X0ljM:-825035343) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L491)):*
`copy` is a reserved word; can you avoid it? Also, this might be simpler as a standalone method with a value receiver.

---

*[sql/testdata/grant_table, line 14 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8r9_vxbvixYqwW52:-KFo8r9_vxbvixYqwW53:-938683042) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/testdata/grant_table#L14)):*
does this really belong in this commit? i suspect it really belongs in the previous one.

---

*[sql/testdata/multi_statement, line 62 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8ussTzTVuRmlx4fp:-KFo8ussTzTVuRmlx4fq:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/testdata/multi_statement#L62)):*
ditto

---

*[sql/testdata/table, line 124 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8wuDUeVQ0rU3k-oZ:-KFo8wuDUeVQ0rU3k-o_:-469765787) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/testdata/table#L124)):*
ditto

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFo9QdTfzz9m4rT8JBz:-1463602527)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
I am not entirely sure what the reasoning behind the first commit is? We no longer have `@` indirection in `qnames` (maybe the commit msg just needs to be updated)

---

Review status: all files reviewed at latest revision, 27 unresolved discussions, all commit checks successful.

---

*[sql/parser/expr.go, line 286 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoKE_Wvd648RQEZp_R:-KFoKE_Wvd648RQEZp_S:-1562708769) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L286)):*
I wonder if it would make things cleaner to just have two versions of `Base` and `Indirect` (one original one normalized).

---

*[sql/parser/expr.go, line 492 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoJovT0XKnoHygv0CK:-KFoJovT0XKnoHygv0CL:-1596278646) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L492)):*
`append(Indirection(nil), n.Indirect...)`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFoJb7JhgUNF0E9IZsI:-438441050)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Reviewed 2 of 2 files at r1, 14 of 17 files at r2, 1 of 5 files at r4, 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 27 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
@raduberinde the first commit was indeed written while we still had @ in qnames. Now there's no longer a particular point to it , but it still seems like a good change since it's been written - keeping track of the original qname is more flexible than keeping just the original string.

---

Review status: 0 of 21 files reviewed at latest revision, 27 unresolved discussions, some commit checks pending.

---

*[server/admin_test.go, line 328 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo938LuOfb0nB5g1Du:-KFoRXgJQ8xfntwCM1hY:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/86354684a992a0970e7cf46962daea0ca6307eda/server/admin_test.go#L328)):*
Done.

---

*[server/admin_test.go, line 344 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo9C0cAs7KWSpku4oQ:-KFoRWNmL9HYv1CxR2XQ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/server/admin_test.go#L344)):*
Done.

---

*[sql/create.go, line 104 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo7X3wBTzejvFBPmQu:-KFoRm3v4120jYXSiezu:-163552779) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/create.go#L104)):*
I don't think everything needs to be mentioned. It's a refactoring commit, part of a refactoring PR... 

---

*[sql/database.go, line 30 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo65UUSC1FDiVYjfJZ:-KFoh7rrv2bdRngBux-x:677588487) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L30)):*
done

---

*[sql/database.go, line 94 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6JvzWwuhHb7jtXFC:-KFoSIhb32EFHxCtfYyd:237700024) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L94)):*
What do you mean method "proliferation"? `getDatabaseDesc` already existed. I'm adding a new version of it, but pretty much deprecating the old one.

isn't this a better API? Why return a `bool` and leave an uninitialized proto around?
I think the take-an-input-proto style API was copied over from some KV methods that need to read a generic proto and whose type they don't know, so they need the caller to allocate it. Otherwise, I think that's a worse style. 

---

*[sql/database.go, line 143 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6jrqNZgp87e-7MMB:-KFoT-p8xusXqrFfg3-N:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L143)):*
Done.

---

*[sql/database.go, line 148 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6mAyY86h9ggdT9fW:-KFoT0KFN-Q876s9rbCa:-1502681677) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L148)):*
isn't this the textbook case for naked returns? The names for the ret values serve as documentation in the function declaration; if I were to repeat the names in the return statement, I run the risk of getting the order wrong.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFoTPRRNipXr_ejNBb4:1133182766) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
well because technically the descriptor exists (or at least its name exists).

---

*[sql/lease.go, line 227 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo9kxqPglbfr9PyXfO:-KFoUPjb55OoaiXQuzJJ:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L227)):*
Done.

---

*[sql/lease.go, line 243 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoAIDA3W_5j3tv4B13:-KFoe8q8LdsqxzbQBN1O:-1830674095) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L243)):*
I'd turn the question around and ask if there's any reason to hoist it? By default, you'd want it declared closer to where it's used, wouldn't you?

These retries are not expected in the common case, and also every retry is a very heavyweight thing - it does the `waitForOneVersion` call which polls the database, then runs a transaction. So I don't think optimizing one allocation is worth much.

---

*[sql/lease.go, line 293 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFk8jLYF2tgRHvVWB3C:-KFoc23oCHxe4SbJjKOz:-1998179453) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L293)):*
you mean a switch? Done.

---

*[sql/structured.go, line 640 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8DscxhwvdL-aMy5h:-KFogWnqNE2A_MS6e0Pz:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L640)):*
Done.

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFof4-TIRUlR7bk3ibj:929326420) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
"upVersion" is a meaningful name - the `up_version` field is commented on the proto. This function doesn't do much now (it used to be coupled with `incrementMutationID` but then I split them, so I probably wouldn't have gone through the trouble of writing it, but since it's written, I think it's a valuable enough accessor. Generally, accessors, particularly mutators, are better than just accessing fields directly, aren't they (easier to grep for, easier to trap)?

---

*[sql/table.go, line 50 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8L1lTbqX7tGcwY0h:-KFop--mLdIT-5x-rH0A:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L50)):*
Done.

---

*[sql/table.go, line 227 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8PdXRsSDfIAiGib9:-KFojKjZBVE_6Nhpifze:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L227)):*
Done.

---

*[sql/table.go, line 260 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8Vs4W-mdWDdjBLMN:-KFoiUsDlDMQAidosIkG:-2080215079) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L260)):*
well I'd love to get rid of this method (and keep `getTableDescEx`), but it's used in a bunch of places. Hopefully they'll migrate over time to the new one and we can delete this one.

---

*[sql/truncate.go, line 57 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8dtvnRSbmWew19aX:-KFokJ-c44dVl6E0MaO9:863923021) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/truncate.go#L57)):*
the function will be reused in another PR

---

*[sql/parser/expr.go, line 286 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoKE_Wvd648RQEZp_R:-KFouCDjD6xYaoDJFMn3:718933465) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L286)):*
i dunno... I'd just keep it as written. Maybe you start off with a normalized qname, so you never have un-normalized base and indirect.

---

*[sql/parser/expr.go, line 482 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo5hKJlLnoWK895xWU:-KFonldjpjrVBzrLxfZ0:-1576686677) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L482)):*
unexported.
I have a deepCopy now, but not sure how it affects this function?

---

*[sql/parser/expr.go, line 491 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo5OUyckwv9FWYYrE1:-KFon_hcm-9fo8nLtd4I:-1076177193) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L491)):*
replaced with a deepCopy. Is this what you had in mind?

---

*[sql/parser/expr.go, line 492 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoJovT0XKnoHygv0CK:-KFonZhNXXx7YSAv2f-3:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/a6cf9e23b1664dffd14c3fc38c27fd361094e3b7/sql/parser/expr.go#L492)):*
Done.

---

*[sql/testdata/grant_table, line 14 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8r9_vxbvixYqwW52:-KFolcTNU0t9eK6I_V2A:-1855997232) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/testdata/grant_table#L14)):*
the tests pass after the previous commit. The change is needed because something in this 2nd commit changed how these errors are generated.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFotWw7klRvzgW6rVC2:851305950)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
It makes things a bit more complicated and adds allocations, we should keep it only if we envision an actual use case for it..

---

Review status: 0 of 21 files reviewed at latest revision, 27 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFoytk-l-RWvV4ASetz:1690232119)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
ok, removed the first commit

---

Review status: 0 of 19 files reviewed at latest revision, 25 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFp2aJsebKng8bmBskO:502858635)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 15 of 17 files at r7, 4 of 4 files at r10.
Review status: all files reviewed at latest revision, 26 unresolved discussions, some commit checks pending.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFozzblFxvej1XwoQOO:542315866) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
I can't see how this semantic can be of use to a caller (and AFAICT we aren't using them in this change). If we get an error in the initial `Get()` we don't know for sure if the descriptor exists or not, and we return false there. So if we get an error, the bool can't really be relied upon.

---

*[sql/table.go, line 263 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp0T9YDIST0VXR2KiG:-KFp0T9YDIST0VXR2KiH:337373439) ([raw file](https://github.com/cockroachdb/cockroach/blob/c50dfec919b3a9e727d3ade2578021efcebb780e/sql/table.go#L263)):*
this can be one line, fits in 100 columns

---

*[sql/testdata/grant_table, line 14 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8r9_vxbvixYqwW52:-KFp1uN0hGuMwC5dua7_:629247306) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/testdata/grant_table#L14)):*
Nice, I see - now the error is generated at a higher layer where we have the full `qname`, rather than inside `getDescriptor` where we only had the table name.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFp328FFEWdZI_ySr93:1417088721)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: all files reviewed at latest revision, 22 unresolved discussions, some commit checks pending.

---

*[sql/lease.go, line 294 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3j7AIuQhOy3Vxj_M:-KFp3j7AIuQhOy3Vxj_N:-2141374080) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c8a682592487581bc7bd90f28d11858e9d46d05/sql/lease.go#L294)):*
Ha, had to look up if `continue` has some special meaning inside a switch.. Seems like it would be equivalent to just leave an empty line, maybe with a comment `// Retry`.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 2 of 21 files at r6, 15 of 17 files at r7, 6 of 6 files at r9, 4 of 4 files at r10.
Review status: all files reviewed at latest revision, 20 unresolved discussions, all commit checks successful.

---

*[server/admin_test.go, line 341 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3UyV6MDgInjXiIkS:-KFp3UyZxfAktKnaSNGJ:29099476) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb35881f25c5a441b719a8b481fe55909315e6e6/server/admin_test.go#L341)):*
"table" still got dropped here?

---

*[sql/alter_table.go, line 204 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3dQz7z3gZ0YiKJiK:-KFp3dQz7z3gZ0YiKJiL:683519995) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/alter_table.go#L204)):*
why is this better than the old code? method calls below look good, but this early bail doesn't seem better.

---

*[sql/alter_table.go, line 220 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3uOXOKBwrVUuDLuc:-KFp3uOXOKBwrVUuDLud:232928688) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/alter_table.go#L220)):*
why `=` over `:=`?

---

*[sql/create.go, line 104 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo7X3wBTzejvFBPmQu:-KFow6xqJdAiT2DHBuOh:1470407108) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/create.go#L104)):*
Right, but you split things up into several commits, so it is very surprising when one of those three commits is 3 times larger than the others and includes a bunch of unmentioned changes.

---

*[sql/database.go, line 94 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6JvzWwuhHb7jtXFC:-KFp41W20wQVO0HmNgGv:1422794580) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L94)):*
Deprecation is something we do when we can't atomically update everything; in this case, you can update all callers. If you intend for this method to replace the other, it's only fair for you to take the burden of update upon yourself.

---

*[sql/database.go, line 148 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6mAyY86h9ggdT9fW:-KFp4KoOAChtS3wTKqML:-45785311) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L148)):*
My position is that naked return is never a good idea.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFp4ZrYulH3peFOQfLh:1937776593) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
What @raduberinde said.

---

*[sql/descriptor.go, line 152 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp4d5H9V4nPoz-Do7e:-KFp4d5H9V4nPoz-Do7f:-1169827610) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/descriptor.go#L152)):*
nope, this should be NewErrorf - this is indicative of a bug, not user error.

---

*[sql/lease.go, line 243 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoAIDA3W_5j3tv4B13:-KFp4q1BAgO4V-740N2k:-2092964781) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L243)):*
Huh? You've already it hoisted it some. Why stop in middle?

---

*[sql/lease.go, line 294 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3j7AIuQhOy3Vxj_M:-KFp6fVPhcLcVi2JIjPC:5172523) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c8a682592487581bc7bd90f28d11858e9d46d05/sql/lease.go#L294)):*
+1

---

*[sql/lease.go, line 298 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp6idn-yZ1wr0qWfLc:-KFp6ido7UDy1nmOs_78:-1110143854) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c8a682592487581bc7bd90f28d11858e9d46d05/sql/lease.go#L298)):*
i think returning `desc, pErr` would be just fine here

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFp5AvPKl12PkV6_A6Y:804378450) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
I would argue that the opposite is true. You haven't made it impossible to access the field directly, so now I just have one extra thing to grep for. Which is better?

---

*[sql/structured.go, line 645 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp5LQ0rUJSliH4rsNk:-KFp5LQ1ZFknDbVvu7N0:1357126384) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/structured.go#L645)):*
It's surprising that a method with this name has a return value at all.

---

*[sql/table.go, line 260 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8Vs4W-mdWDdjBLMN:-KFp5WmpD8avgRuyu3GH:-1140499301) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L260)):*
Again, I don't think this is fair (and nobody else does this). If you're going to assert that this API is better than the existing, please replace all callers.

```
$ git grep getTableDesc | wc -l
      22
$ git grep getDatabaseDesc | wc -l
      15
```

Doesn't seem so onerous.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

andreimatei committed:
sql: get rid of GetDatabaseDescEx

andreimatei commented:
Review status: 8 of 22 files reviewed at latest revision, 17 unresolved discussions, some commit checks pending.

---

*[server/admin_test.go, line 341 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3UyV6MDgInjXiIkS:-KFpEpvOF5jNnt2RZpKx:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/eb35881f25c5a441b719a8b481fe55909315e6e6/server/admin_test.go#L341)):*
Done.

---

*[sql/alter_table.go, line 204 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3dQz7z3gZ0YiKJiK:-KFpGANUztpfKwPxkpaI:1342529493) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/alter_table.go#L204)):*
I find it more readable than that `return` in the `else if`.

---

*[sql/alter_table.go, line 220 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3uOXOKBwrVUuDLuc:-KFpGYsL04bv9czVlxkR:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/alter_table.go#L220)):*
Done.

---

*[sql/database.go, line 94 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6JvzWwuhHb7jtXFC:-KFppjsc9QFqYZeyI1Ri:-578609703) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/database.go#L94)):*
ok, added a new commit getting rid of one of the 2

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFpHKSp21G3ba70QK3Z:-1345194033) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
I agree that if there's an error, the `bool` shouldn't matter much. So you can argue that we can return any value here (e.g. `true`).

If we're bikeshedding, the `false` is ambiguous about the existence, but the `true` value is not. 
You can argue that we might as well return a consistent value everywhere, and `false` seems a more natural choice. But then would you also change the `true`s below?

---

*[sql/descriptor.go, line 152 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp4d5H9V4nPoz-Do7e:-KFpovL-kBjblHzGbULg:677588487) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/descriptor.go#L152)):*
done

---

*[sql/lease.go, line 243 \[r3\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFoAIDA3W_5j3tv4B13:-KFppIv4fWFtCJjtuF9U:-997890763) ([raw file](https://github.com/cockroachdb/cockroach/blob/919d0d48894e63e2654a891c2511c079d7bf2228/sql/lease.go#L243)):*
I've hoisted it so that it escapes the closure, so I can return it from this function.

---

*[sql/lease.go, line 294 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp3j7AIuQhOy3Vxj_M:-KFpqehgL5ttxZeDjYre:1449906964) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c8a682592487581bc7bd90f28d11858e9d46d05/sql/lease.go#L294)):*
but but `continue` makes the reader not have to read the rest of the body of the loop to see if anything else happens.
but ok done.

---

*[sql/lease.go, line 298 \[r10\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp6idn-yZ1wr0qWfLc:-KFpqyb6pPU0xS5OyMQC:159024384) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c8a682592487581bc7bd90f28d11858e9d46d05/sql/lease.go#L298)):*
I had it like that and I felt the need for a comment to say that it handles the `pErr == nil` case, so I just rewrote it this way for clarity.
I generally find it surprising that `pErr.GetDetail` works on a `pErr == nil`, but I guess that ship has sailed.

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFprLyw0dL6_nLXYh3Q:-892961906) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
Well I can't technically make it impossible to access the field directly unfortunately. But this field is kind of a big deal, so hopefully everybody who wants to use it will look around to what others are doing.

---

*[sql/structured.go, line 645 \[r9\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp5LQ0rUJSliH4rsNk:-KFpre7NTWVD1HuxFi0n:-403841541) ([raw file](https://github.com/cockroachdb/cockroach/blob/6906eee20de4f51171e9a63d37df0edba52e9486/sql/structured.go#L645)):*
well I don't have a better idea. `getNextMutationID` is technically wrong, since this returns the previous mutation id. And you didn't like `incrementMutationID`.

---

*[sql/table.go, line 260 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo8Vs4W-mdWDdjBLMN:-KFppXUDhwXTqj_rNoP7:-199150218) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/table.go#L260)):*
ok, got rid of the old API in a new commit

---

*[sql/table.go, line 263 \[r7\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFp0T9YDIST0VXR2KiG:-KFppVFUYqU892o4fk6W:-499146009) ([raw file](https://github.com/cockroachdb/cockroach/blob/c50dfec919b3a9e727d3ade2578021efcebb780e/sql/table.go#L263)):*
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
new commit LGTM

---

Reviewed 21 of 21 files at r6, 1 of 6 files at r11, 9 of 9 files at r12, 4 of 4 files at r13.
Review status: all files reviewed at latest revision, 16 unresolved discussions, all commit checks successful.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFpw8rsBV2UhNY8lXw3:1566407175) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
Yes, I would change those to false as well. This is the semantic we are using in most places and with which everyone should be accustomed to: if there is an error any other returns are zero and don't mean anything. No point in making things more complicated if that is sufficient.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFpySL1CSw149igSz80:1132795992)*
<!-- Sent from Reviewable.io -->

tamird commented:
new commit LGTM but should be squashed with preceding

---

Reviewed 6 of 6 files at r11, 9 of 9 files at r12, 4 of 4 files at r13.
Review status: all files reviewed at latest revision, 8 unresolved discussions, all commit checks successful.

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFq0i0mEYUCzneROzxF:-1427599940) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
What's the upshot again? In my experience trivial methods like this end up complicating finding usage of a field. For instance, this will break the ability to use `oracle referrs` with this field.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-:-KFq1gah7YefRtX7jPkD:-1608190412)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
Review status: 8 of 22 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFq36vP53fQr6RtyKm3:1828734116) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
Done.
Look what happened to the final `return` though. It's probably for the better.

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFq5okvYsX0EnXNBv31:-858949407) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
ok removed

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
Review status: 8 of 22 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[sql/structured.go, line 649 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo85LCUOUw57aX_phF:-KFq9rpsjGX4hX8Pi9yl:-442100400) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/structured.go#L649)):*
Oh I just remembered why this was a thing when rebasing another PR - this function evolves to have a check in it (checking that you don't `setUpVersion` after the table has been marked for deletion), so I think it'll come back.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

RaduBerinde commented:
Review status: 8 of 22 files reviewed at latest revision, 8 unresolved discussions, some commit checks pending.

---

*[sql/descriptor.go, line 145 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160#-KFo6wl3lW3PxOVd9TfS:-KFqAfO8Gz6AE5OeZnj7:4552694) ([raw file](https://github.com/cockroachdb/cockroach/blob/0e8f00ab4d7c1ab92492bd35e92bf7379606a771/sql/descriptor.go#L145)):*
LGTM, thanks

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 14 files at r14, 9 of 9 files at r15, 4 of 4 files at r16.
Review status: all files reviewed at latest revision, 7 unresolved discussions, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/6160)*
<!-- Sent from Reviewable.io -->
